rmw_zenoh_cpp/src/zenohd/main.cpp

```diff
@@ -63,7 +64,7 @@ int main(int argc, char ** argv)
   // If not already defined, set the logging environment variable for Zenoh router
   // to info level by default.
   // TODO(Yadunund): Switch to rcutils_get_env once it supports not overwriting values.
-  if (setenv(ZENOH_LOG_ENV_VAR_STR, ZENOH_LOG_INFO_LEVEL_STR, 0) != 0) {
+  if (rcutils_set_env_overwrite(ZENOH_LOG_ENV_VAR_STR, ZENOH_LOG_INFO_LEVEL_STR, 0) != 0) {
```

Comment:
It might be better to leave the switch to `rcutils_set_env_overwrite/rcutils_set_env2` once the upstream PR is reviewed and merged. Given that it may not be backported to iron/jazzy, it might be best if we use a combination of `rcutils_get_env` check first followed by `rcutils_set_env` to get the same behavior as `rcutils_set_env_overwrite`.